aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -830,6 +830,83 @@ For more information on JavaScript isolation with JavaScript modules, see <xref:
 
 :::moniker-end
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Hashed routing to named elements
+
+Navigate to a hashed (`#`) named element using the following approaches:
+
+* Anchor element (`<a>`) with an `href`.
+* <xref:Microsoft.AspNetCore.Components.Routing.NavLink> component with an `href`.
+* <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A?displayProperty=nameWithType> passing the URL.
+
+The following example demonstrates hashed routing to named H2 headings.
+
+In the `Index` (`Pages/Index.razor`) and `Counter` (`Pages/Counter.razor`) components, place the following markup at the bottom of the existing component markup:
+
+```razor
+<div class="border border-info rounded bg-info" style="height:500px"></div>
+
+<h2 id="targetElement">Target H2 heading</h2>
+<p>Content!</p>
+```
+
+Add the following `HashedRouting` component to the app:
+
+`Pages/HashedRouting.razor`:
+
+```razor
+@page "/hashed-routing"
+@inject NavigationManager Navigation
+
+<PageTitle>Hashed routing</PageTitle>
+
+<h1>Hashed routing to named elements</h1>
+
+<ul>
+    <li>
+        <a href="#targetElement">
```

Comment:
I changed this behavior recently because it was incorrect. Here it should be `href="/hashed-routing#targetElement"`. There is no scrolling to the element on the same page when clicking on `<a href="#..."></a>`. https://github.com/dotnet/aspnetcore/issues/47687

Reply:
```suggestion
        <a href="hashed-routing#targetElement">
```

Reply:
yes, like this